pkg/daemon/controller/controller.go

```diff
@@ -425,49 +480,65 @@ func (c *Controller) handleVxlanInterfaceNeighEvent() error {
 		}
 	}
 
-	ch := make(chan netlink.NeighUpdate, NeighUpdateChanSize)
-
-	// clear stale neigh entries for vxlan interface at the first time
-	linkList, err := netlink.LinkList()
-	if err != nil {
-		return fmt.Errorf("list link failed: %v", err)
-	}
-
-	for _, link := range linkList {
-		if strings.Contains(link.Attrs().Name, containernetwork.VxlanLinkInfix) {
-			if err := neigh.ClearStaleNeighEntries(link.Attrs().Index); err != nil {
-				return fmt.Errorf("clear stale neigh entries for link %v failed: %v",
-					link.Attrs().Name, err)
-			}
-		}
-	}
-
 	hostNetNs, err := netns.Get()
 	if err != nil {
 		return fmt.Errorf("get root netns failed: %v", err)
 	}
 
-	if err := netlink.NeighSubscribeAt(hostNetNs, ch, nil); err != nil {
-		return fmt.Errorf("subscribe neigh update event failed: %v", err)
-	}
+	timer := time.NewTicker(c.config.VxlanExpiredNeighCachesClearInterval)
+	errorMessageWrapper := initErrorMessageWrapper("handle vxlan interface neigh event failed: ")
 
 	go func() {
-		errorMessageWrapper := initErrorMessageWrapper("handle vxlan interface neigh event failed: ")
+		for {
+			// Clear stale and failed neigh entries for vxlan interface at the first time.
+			// Once neigh subscribe failed, it neigh fail because of "No buffer space available" problem,
+			// need to clean expired neigh caches
+			if err := clearVxlanExpiredNeighCaches(); err != nil {
```

Comment:
caches clear may not be in a loop

Reply:
As it says in comments, the goroutine of `netlink.NeighSubscribe` might exit most probably because of neigh caches buffer space error. In this situation, we are supposed to clean the expired neigh caches once in case it fails over and over again, while normally it only work once if the goroutine of `netlink.NeighSubscribe` never exit.

Reply:
got it!